Bots/JavaScript/Consumers/CodeFirst/WaterfallHostBot/skillsConfiguration.js

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { EchoSkill } = require('./skills/echoSkill');
+const { WaterfallSkill } = require('./skills/waterfallSkill');
+const { TeamsSkill } = require('./skills/teamsSkill');
+const { SkillDefinition } = require('./skills/skillDefinition');
+
+/**
+ * A helper class that loads Skills information from configuration.
+ */
+class SkillsConfiguration {
+    constructor() {
+        this.skillsData = {};
+
+        var skillVariables = Object.keys(process.env).filter(prop => prop.startsWith('skill_'));

```

Comment:
`var` should be `const` or `let`

I think functional style is easier to read and reason about, but this is just a suggestion. No need to change this if you disagree.

```javascript
this.skillsData = Object.entries(process.env).filter(([key]) => key.startsWith('skill_')).reduce((acc, [key, value]) => {
  const [, id, attr] = key.split('_');

  acc[id] = acc[id] || { definition: this.createSkillDefinition({ id }) };

  const propName = { appid: 'appId', endpoint: 'skillEndpoint' }[attr.toLowerCase()];
  if (!propName) { throw new Error(...) }

  acc[id][propName] = value;

  return acc;
}, {});
```